### PR TITLE
riot-desktop: wrap with wrapGAppsHook

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
By not using wrapGAppsHook we're missing other important things in the wrapper.
Things like
```
export XDG_DATA_DIRS='/nix/store/h11fidilg14qg0ppx7dsg1na2bybpk8c-riot-desktop-1.5.15/share'${XDG_DATA_DIRS:+':'}$XDG_DATA_DIRS
export GIO_EXTRA_MODULES='/nix/store/z7gzw2bb3ypvf1wq1fjjn5r45idfgg0g-dconf-0.34.0-lib/lib/gio/modules'${GIO_EXTRA_MODULES:+':'}$GIO_EXTRA_MODULES
export GDK_PIXBUF_MODULE_FILE='/nix/store/18l8ly5235r94s2rr122i3cfichccxqn-librsvg-2.46.4/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache'
```

dconf is needed for gsettings.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by pacien — APPROVED** [review]
LGTM; successfully tested the selection of a file.

This might need to be backported to the `release-20.03` branch.

File: pkgs/applications/networking/instant-messengers/riot/riot-desktop.nix [hunk 1]
```diff
@@ -1,7 +1,4 @@
-{ stdenv, fetchFromGitHub
-, makeWrapper, makeDesktopItem, mkYarnPackage
-, electron_7, riot-web, gtk3,
-}:
+{ pkgs, stdenv, fetchFromGitHub, makeWrapper, makeDesktopItem, electron_7, riot-web, mkYarnPackage, wrapGAppsHook, glib, gtk3 }:
```
Comment (rnhmjoj) [inline pkgs/applications/networking/instant-messengers/riot/riot-desktop.nix:1]: I don't really like this: it makes the diff really hard to read. :(

File: pkgs/applications/networking/instant-messengers/riot/riot-desktop.nix [hunk 2]
```diff
@@ -27,7 +24,14 @@ in mkYarnPackage rec {
   packageJSON = ./riot-desktop-package.json;
   yarnNix = ./riot-desktop-yarndeps.nix;
 
-  nativeBuildInputs = [ makeWrapper gtk3 ];
+  nativeBuildInputs = [ wrapGAppsHook ];
+
+  extraBuildInputs = [
```
Comment (rnhmjoj) [inline pkgs/applications/networking/instant-messengers/riot/riot-desktop.nix:29]: Oh, right: gtk3 is not a native dependency. I don't know what `extraBuildInputs` is exactly, though.

Comment (worldofpeace) [reply pkgs/applications/networking/instant-messengers/riot/riot-desktop.nix:29]: mkYarnPackage doesn't have buildInputs exposed apparently.

File: pkgs/applications/networking/instant-messengers/riot/riot-desktop.nix (hunk 1 above)
Comment (worldofpeace) [reply pkgs/applications/networking/instant-messengers/riot/riot-desktop.nix]: you can diff on github without whitespace changes.
and I realized I didn't intend to change this.

**Review by rnhmjoj — APPROVED** [review]
Looks good. Thank you for fixing my original PR.

**Comment by worldofpeace** [comment]
No problem @rnhmjoj. 
Backported to 20.03 in 57ecbc5c61be2bb387eaa1802ecf217b8a21bf63 a128dd3af8d5d8c91eb153e9976ceb3a39c7a868.
We probably should do the same for 19.09. (but it's pretty out of date there).

**Pull request merged**